draft-chen-cfrg-vdaf-pine.md

```diff
@@ -220,46 +222,148 @@ does not wrap around the field modulus. However, this approach has relatively
+> to go into full detail yet. The goal is to give the reader an intuition for
+> what the circuit will do, including the bit checks, norm computation, and
+> wraparound checks. Be sure to emphasize that the wraparound test results are
+> not transmitted: instead the Aggregators derive shares of the tests results
+> by running the wraparound tests themselves.
+
+### Encoding of Floating Point Numbers into Field Elements
+
+> TODO Specify how floating point numbers are represented as field elements.
+
+### Encoding the Range-Checked, Squared Norm
+
+> TODO Specify how the Client encodes the norm such that the Aggregators can
+> check that it is in the desired range.
+
+### Running the Wraparound Tests
+
+> XXX Give an overview of how the wraparound test results are generated from
+> the XOF seed.
+
+### Encoding the Range-Checked, Wraparound Check Results
+
+> TODO Specify how the Client encodes the result of each wraparound check such
+> that the Aggregators can check that each is in the desired range.
+
+## The FLP Circuit
+
+> XXX Give an overview of the circuit's operation. Here it would be appropriate
+> to name all of the intermediate results: the bit checks, the wraparound
+> checks result (the reduction over all of the checks), the wraparound success
+> count, the squred norm equality test, and the squared norm range check.
+
+> TODO Specify the implementation of `Valid` from {{Section 7.3.2 of !VDAF}}.
+
+# The PINE VDAF {#vdaf}
+
+This section specifies the complete VDAF.
+
+> XXX Give a high level overview of how the encoding scheme and FLP are used by
+> PINE. Here we want to define the the "wraparound joint randomness" and the
+> "verification joint randomness" and give some details of how each is derived.
+> We also want to mention the multiproof feature and the relationship between
+> the number of proofs, the field size, and soundness.
+>
+> Remember, keep it short and intuitive. The full specificaiton of the
+> algorithms will come later, in a future draft. If you're using too many
+> variable names, then you're probably not on the right track.
+
+## Sharding
+
+> TODO Specify the implementation of `Vdaf.shard()`.
+
+## Preparation
+
+> TODO Specify the implementations of `Vdaf.prep_init()`,
+> `.prep_shares_to_prep()`, and `.prep_next()`.
+
+## Aggregation
+
+> TODO Specify the implementation of `Vdaf.aggregate()`.
+
+## Unsharding
+
+> TODO Specify the implementation of `Vdaf.unshard()`.
+
+# Variants
+
+> TODO Specify concrete parameterizations of VDAFs, including the choice of
+> field, number of proofs, and valid ranges for the parameters in
+> {{pine-user-param}}.
+
+# Security Considerations
+
+Our security considerations for PINE are the same as those for Prio3 described
+in {{Section 9 of !VDAF}}.
+
+> XXX Given that we can tune the parameters such that the ZK error is
+> negligible, I don't think this needs to be mentioned here.
+>
+> Is there anything else worth mentioning?
```

Comment:
The other difference from Prio3 is there are two Fiat-Shamir derivation, so the robustness needs to account for the random oracle attack from both wraparound joint randomness and verification joint randomness?

Reply:
Our goal will be to tune the parameters such that the user doesn't have to worry about this.